From f87652e89d5f3051f95245b1bed2e1f75e43b2f9 Mon Sep 17 00:00:00 2001
From: Jian-Hong Pan <jhp@endlessos.org>
Date: Tue, 15 Apr 2025 14:17:54 +0800
Subject: [PATCH 1/3] [Endless] linux-pam: Allow weak passwords

Allow weak passwords by dropping pam_pwquality.so module check and
pam_unix.so's use_authtok. If pam_unix.so keeps use_authtok, the new
password still have to pass pam_pwquality.so test [1]. Here is an
example of dropped pam_pwquality.so module, but pam_unix.so keeping
use_authtok. Passwd prints "Authentication token manipulation error"
directly:

$ passwd
Changing password for dev.
Current password:
passwd: Authentication token manipulation error
passwd: password unchanged

Besides, the default nullok_secure [2] setting only allows passwordless
accounts to access services like sudo and polkit from certain types of
terminals. Instead, nullok is configured here because EOS would like
passwordless accounts to be usable in all contexts, including terminal
apps running in the desktop environment.

AC:
- Command "passwd" can change current user's password to 111.
- When logged into the desktop of passwordless account, running
  "sudo -s" in a Terminal gives a root shell without a password prompt.
- When logged into the desktop of passwordless account, in Settings,
  choosing Users then clicking Unlock unlocks the panel without a
  password prompt.

[1]: https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/7/html/system-level_authentication_guide/pam_configuration_files#Sample_PAM_Configuration_Files
[2]: https://manpages.debian.org/stretch/libpam-modules/pam_unix.8.en.html

https://phabricator.endlessm.com/T14095
https://phabricator.endlessm.com/T35802
---
 files/linux-pam-config/password-auth | 3 +--
 files/linux-pam-config/system-auth   | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/files/linux-pam-config/password-auth b/files/linux-pam-config/password-auth
index a2803b492..1df8aac70 100644
--- a/files/linux-pam-config/password-auth
+++ b/files/linux-pam-config/password-auth
@@ -7,8 +7,7 @@ auth        required      pam_deny.so
 account     required      pam_unix.so
 
 -password   sufficient    pam_systemd_home.so
-password    requisite     pam_pwquality.so try_first_pass local_users_only retry=3 authtok_type=
-password    sufficient    pam_unix.so try_first_pass use_authtok nullok_secure sha512 shadow
+password    sufficient    pam_unix.so try_first_pass nullok sha512 shadow
 password    required      pam_deny.so
 
 -session    optional      pam_systemd_home.so
diff --git a/files/linux-pam-config/system-auth b/files/linux-pam-config/system-auth
index a2803b492..1df8aac70 100644
--- a/files/linux-pam-config/system-auth
+++ b/files/linux-pam-config/system-auth
@@ -7,8 +7,7 @@ auth        required      pam_deny.so
 account     required      pam_unix.so
 
 -password   sufficient    pam_systemd_home.so
-password    requisite     pam_pwquality.so try_first_pass local_users_only retry=3 authtok_type=
-password    sufficient    pam_unix.so try_first_pass use_authtok nullok_secure sha512 shadow
+password    sufficient    pam_unix.so try_first_pass nullok sha512 shadow
 password    required      pam_deny.so
 
 -session    optional      pam_systemd_home.so
-- 
2.49.0

